pkg/controller/reconciler/annotations.go

```diff
@@ -98,9 +107,11 @@ func readAnnotations(deployment *appsv1.Deployment) (deploymentAnnotations, erro
 		if err != nil {
 			return result, fmt.Errorf("failed to unmarshal status annotation: %w", err)
 		}
-	}
 
-	result.Status = &s
+		// The reason I moved this here is because if Status is not nil,
+		// then we should set it in the result. Otherwise it should be nil.
+		result.Status = &s
```

Comment:
The problem here was that the previous place of this line, which was outside of this **if case**, was causing two different Status: nil and empty object (**deploymentStatus{}**). And in other parts of the code we are checking is Status is nil not if Status is an empty object.